plugin/src/main/java/com/denizenscript/denizen/objects/properties/material/MaterialSwitchable.java

```diff
@@ -0,0 +1,129 @@
+    }
+
+    public static MaterialSwitchable getFrom(ObjectTag _material) {
+        if (!describes(_material)) {
+            return null;
+        }
+        else {
+            return new MaterialSwitchable((MaterialTag) _material);
+        }
+    }
+
+    public static final String[] handledMechs = new String[] {
+            "switched"
+    };
+
+    private MaterialSwitchable(MaterialTag _material) {
+        material = _material;
+    }
+
+    MaterialTag material;
+
+    public static void registerTags() {
+
+        // <--[tag]
+        // @attribute <MaterialTag.switched>
+        // @returns ElementTag(Boolean)
+        // @group properties
+        // @description
+        // Returns if this material is switched, or not. (doors, dispensers, and other redstone materials).
+        // -->
+        PropertyParser.<MaterialSwitchable>registerTag("switched", (attribute, material) -> {
+            return new ElementTag(material.getState());
+        });
+    }
+
+    public boolean isPowerable() {
+        return material.getModernData().data instanceof Powerable;
+    }
+
+    public Powerable getPowerable() {
+        return (Powerable) material.getModernData().data;
+    }
+
+    public boolean isOpenable() {
+        return material.getModernData().data instanceof Openable;
+    }
+
+    public Openable getOpenable() {
+        return (Openable) material.getModernData().data;
+    }
+
+    public Dispenser getDispenser() {
+        return (Dispenser) material.getModernData().data;
+    }
+
+    public boolean getState() {
+        if (isOpenable()) {
+            return getOpenable().isOpen();
+        }
+        else if (isPowerable()) {
+            return getPowerable().isPowered();
+        }
+        else {
+            return getDispenser().isTriggered();
+        }
+    }
+
+    public void setState(boolean state) {
+        if (isOpenable()) {
+            getOpenable().setOpen(state);
+            return;
+        }
+        else if (isPowerable()) {
+            getPowerable().setPowered(state);
+            return;
+        }
+        else {
+            getDispenser().setTriggered(state);
+            return;
+        }
+    }
+
+    @Override
+    public String getPropertyString() {
+        return String.valueOf(getState());
+    }
+
+    @Override
+    public String getPropertyId() {
+        return "switched";
+    }
+
+    @Override
+    public void adjust(Mechanism mechanism) {
+
+        // <--[mechanism]
+        // @object MaterialTag
+        // @name switched
+        // @input ElementTag(boolean)
+        // @description
+        // Sets if this material is switched, or not. (doors, dispensers, and other redstone materials).
+        // @tags
+        // <MaterialTag.switched>
+        // -->
+        if (mechanism.matches("switched") && mechanism.requireBoolean()) {
+            setState(mechanism.getValue().asBoolean());
+        }
+    }
+}
```

Comment:
newline at end of file (this is one of the things that IntelliJ autoformatting should do for you automatically)